src/components/SegmentControl/MarkedSegmentControl.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { View, StyleSheet } from 'react-native';
+import { connect } from 'react-redux';
+import { bool, func } from 'prop-types';
+
+import BaseSegmentControl from '../Generic/BaseSegmentControl';
+import BaseText from '../Generic/BaseText';
+import { toggleShowMarkedOnly } from '../../redux/action-creators';
+
+const allRecordsDescription = 'Displays all records.';
+const highlightedRecordsDescription = 'Only displays highlighted records.';
+
+const MarkedSegmentControl = ({
+  showMarkedOnly,
+  toggleShowMarkedOnlyAction,
+}) => {
+  const segControlIndex = showMarkedOnly ? 1 : 0;
+  const description = segControlIndex === 0 ? allRecordsDescription : highlightedRecordsDescription;
+  const handleChange = (event) => {
+    if (event === 0) {
+      toggleShowMarkedOnlyAction(false);
+    } else {
+      toggleShowMarkedOnlyAction(true);
+    }
+  };
+
+  return (
+    <View style={styles.root}>
+      <BaseSegmentControl
+        values={['All Records', 'Highlighted Records']}
+        selectedIndex={segControlIndex}
+        handleChange={handleChange}
```

Comment:
not 100% of latest conventions, but this possibly should be `onChange={handleChange}`

eg, there's the following ESLint rule: (which may not be configured for this repo)

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-handler-names.md